packages/ui/src/views/pipelines/pipeline-list/pipeline-list-page.tsx

```diff
@@ -0,0 +1,97 @@
+import { ChangeEvent, FC, ReactNode, useCallback, useState } from 'react'
+import { Link } from 'react-router-dom'
+
+import { Button, ListActions, PaginationComponent, SearchBox, Spacer, Text } from '@/components'
+import { debounce } from 'lodash-es'
+
+import { SandboxLayout } from '../../index'
+import { PipelineList } from './pipeline-list'
+import { PipelineListPageProps } from './types'
+
+const LinkComponent = ({ to, children }: { to: string; children: ReactNode }) => <Link to={to}>{children}</Link>
+
+const SandboxPipelineListPage: FC<PipelineListPageProps> = ({
+  usePipelineListStore,
+  useTranslationStore,
+  isLoading,
+  isError,
+  errorMessage,
+  searchQuery,
+  setSearchQuery
+}) => {
+  const { t } = useTranslationStore()
+  const { pipelines, totalPages, page, setPage } = usePipelineListStore()
+
+  const [searchInput, setSearchInput] = useState(searchQuery)
+
+  const debouncedSetSearchQuery = debounce(searchQuery => {
+    setSearchQuery(searchQuery || null)
+  }, 300)
+
+  const handleInputChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+    setSearchInput(e.target.value)
+    debouncedSetSearchQuery(e.target.value)
+  }, [])
+
+  if (isError)
+    return (
+      <>
+        <SandboxLayout.Main hasHeader hasLeftPanel>
+          <SandboxLayout.Content>
+            <Spacer size={2} />
+            <Text size={1} className="text-destructive">
+              {errorMessage || 'Something went wrong'}
```

Comment:
there should be a nodata state here I guess? with description as the errormessage